### ISSUE TITLE
Python Packages 

### ISSUE THREAD

**Issue by tsoiadelina** [vktrbr/optimization_ml#55]
##### Inputs (for each func)
- [x] array of predictors (np.array) (**add assert for type np.ndarray**) 
- [x] array of the predicted variable (**add add assert for type np.ndarray**)
- [x] type of regulators (l1, l2, none)
##### Output (for each func)
- [x] function in analytical form (is it necessary?)
- [x] weights of each x (yea)
- [x] finding a free regression term (is it necessary?)
## Functions
reference: https://scikit-learn.org/stable/modules/linear_model.html#ordinary-least-squares

Loss: 
![image](https://user-images.githubusercontent.com/52676181/162635594-04782393-e582-4e01-9ced-39ad579b215e.png)

L1:
![image](https://user-images.githubusercontent.com/52676181/162635578-28af9f62-bf8d-4649-aac7-550c5eed5cb5.png)

L2: 
![image](https://user-images.githubusercontent.com/52676181/162635567-f5d58212-d426-4848-9603-ce897d93f82b.png)

- [x] Linear regression with L1
- [x] Linear regression with L2
- [x] Polynomial with L1

1. Make polynomial features with [sklearn: PolynomialFeatures](https://scikit-learn.org/stable/modules/generated/sklearn.preprocessing.PolynomialFeatures.html#sklearn.preprocessing.PolynomialFeatures)
2. Make linear regression on this features
- [x] Polynomial with L2

#### exponential: 

Let's make THETA = [THETA0, *THETA1] is m dimensional vector-column. 
And try:
1. ~~minimize sse loss without logarithmic~~ 
![image](https://user-images.githubusercontent.com/52676181/163030863-2273a454-c73f-4da6-b455-d54a2b9e3a82.png)
2. minimize after logarithmic
- [x] Exponential with L1
- [x] Exponential with L2

![image](https://user-images.githubusercontent.com/52676181/162632735-9900c224-0664-43cb-9793-764e0e4247ec.png)

**Comment by tsoiadelina** [comment]
How should I use methods here?

**Comment by vktrbr** [comment]
[algorithms](https://github.com/vktrbr/optimization_ml/tree/main/Regression/algorithms)
